Optional credentials VPN config option

dmlambea opened the pull request:
Fixes #162

This PR makes the credentials (username/password) optional, so that for certificate-only VPN's the connecting procedure no longer asks for username/password.

dmlambea commented:
Hi, @theinvisible 

This PR has been here for a while now. Is there anything I can do to help you reviewing it?  Thanks!

theinvisible commented:
Hi,
sorry for late answer. Can you check please if it still work in latest git source, thanks.

dmlambea commented:
Hi, @theinvisible 

I've just rebased the code onto your latest master and rebuilt my binary.  I'm already connected to my company's VPN and it works flawlessly.

theinvisible commented:
Have tested it now with our vpn profiles.

When there is both username and password set the connect failes because the modified if statement in vpnProcess::startVPN() empties now the given username/password. Can you fix this? Otherwise i will have a look...

dmlambea commented:
@theinvisible you're right, the creds. were emptied every time the dialog wasn't to be shown, making a properly configured user/pass to be cleared out.  The conflicting `else` part is now removed and it should work. I cannot fully test it because my VPN provider is certificate-only but I since get authentication failures when I enable it, I guess the credentials are being sent and it leads to the auth error.

I've rebased again onto your master branch. May you please check to see if it now works for you?

dmlambea commented:
Thank you very much, @theinvisible
